Resolve Symbolics.Num - ForwardDiff.Dual  ambiguities

asterycs opened the pull request:
This is a first attempt at resolving https://github.com/JuliaSymbolics/Symbolics.jl/issues/1032.

The easiest way to resolve the issue seems to be to copy the relevant method definitions from [ForwardDiff](https://github.com/JuliaDiff/ForwardDiff.jl/blob/master/src/dual.jl), which is what is done in this PR.

Unfortunately this introduces a lot of code duplication. An alternative approach would be to convert the method definition loops in https://github.com/JuliaDiff/ForwardDiff.jl/blob/master/src/dual.jl into functions that take a list of the ambiguous types as argument. However, this requires more changes to ForwardDiff in order to prevent existing methods from getting overwritten (e.g. [here](https://github.com/JuliaDiff/ForwardDiff.jl/blob/master/src/dual.jl#L139)).

Let me know if you'd have any suggestions for this PR and I'll accommodate them.

ChrisRackauckas commented:
Looks, good, though just needs some test fixes https://github.com/JuliaSymbolics/Symbolics.jl/actions/runs/7438325797/job/20241004954?pr=1036#step:7:650

asterycs commented:
Thanks @ChrisRackauckas for checking! I updated the tests, should work better now.

I also removed the fma method definitions which I thought I had already deleted.
